### PR TITLE
feat: issue #344 - replaced kubeval with kubeconform

### PR DESCRIPTION
**Pull request by olcuhu**
**All Submissions.**

- [X] Have you followed the guidelines in our [Contributing document](../blob/master/CONTRIBUTING.md)?
- [X] Have you checked to ensure there aren't other open [Pull Requests][pull_requests] for the same update/change?

**Code Submissions.**

- [X] Does your submission pass linting, tests, and security analysis?
acceptance tests failing, will look into that

**Changes to Core Features.**

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
Addresses issue #344, kubeval is not maintained and marks some valid YAML files as invalid. kubeconform is the recommended replacement and seems to work as intended for the YAML file in #315 that was causing the issue.
- [x] Have you written new tests for your core changes, as applicable?

### CODE REVIEW

**Comment by 06kellyjac** [comment]
Thanks a lot for working on this. Ill try take a look at it tomorrow

**Comment by olcuhu** [comment]
No worries - it looks like the failing tests are due to kubeconform being more sensitive than kubeval to missing selectors on deployments etc so I’ll fix the tests tomorrow and let you know when it’s ready to review

**Comment by olcuhu** [comment]
@06kellyjac looks like it's all passing now - I had to locally ignore a couple of the shellcheck rules on the test/helper.bash script as the solutions they were insisting on didn't work. Don't love ignoring test results, but given it's not a production-facing part of the project I can live with it.

File: pkg/ruler/ruleset.go [hunk 1]
```diff
@@ -385,34 +384,26 @@ func (rs *Ruleset) generateReport(fileName string, json []byte, schemaDir string
...
-			report.Message += "This resource is invalid, Kubernetes kind not found"
+	for _, res := range v.Validate(fileName, f) { // A file might contain multiple resources
+		// File starts with ---, the parser assumes a first empty resource
+		if res.Status == validator.Invalid {
+			report.Message += res.Err.Error() + "\n"
+			// fmt.Printf("resource %d in file %s is not valid: %s", i, fileName, res.Err)
+		}
+		if res.Status == validator.Error {
+			report.Message += res.Err.Error()
+			// fmt.Printf("error while processing resource %d in file %s: %s", i, fileName, res.Err)
```
Comment (06kellyjac) [inline pkg/ruler/ruleset.go:406]:
```suggestion
	for _, res := range v.Validate(fileName, f) {
		// A file might contain multiple resources
		// File starts with ---, the parser assumes a first empty resource
		if res.Status == validator.Invalid {
			report.Message += res.Err.Error() + "\n"
		}
		if res.Status == validator.Error {
			report.Message += res.Err.Error()
```

**Review by 06kellyjac — APPROVED** [review]

**Pull request merged**